BYT/ColorManager.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  ColorManager.swift
+//  BYT
+//
+//  Created by Tom Seymour on 1/26/17.
+//  Copyright © 2017 AccessLite. All rights reserved.
+//
+
+import Foundation
+
+class ColorManager {
+    
+    static let shared = ColorManager()
+    
+    var currentColorScheme: ColorScheme! {
+        didSet {
+            self.saveCurrentColorScheme()
+        }
+    }
+    var colorSchemes: [ColorScheme]! {
+        didSet {
+            self.saveColorSchemes()
+        }
+    }
+
+    private static let currentColorSchemeKey: String = "CurrentColorSchemeKey"
+    private static let colorSchemesKey: String = "colorSchemesKey"
+    private static let defaults = UserDefaults.standard
+
+    private init() {}
+    
+    func saveCurrentColorScheme() {
+        do {
+            let currentColorSchemeData: Data = try ColorManager.shared.currentColorScheme.toData()
+            ColorManager.defaults.set(currentColorSchemeData, forKey: ColorManager.currentColorSchemeKey)
+            print("think I successfully saved the currentColorScheme")
+        } catch {
+            print("Error converting currentColorScheme to data")
+        }
+    }
+    
+    func loadCurrentColorScheme() {
+        if let savedDataFromDefaults = ColorManager.defaults.value(forKey: ColorManager.currentColorSchemeKey) as? Data {
+            self.currentColorScheme = ColorScheme(data: savedDataFromDefaults)
+        }
```

Comment:
this crashes due to explicitly unwrapping on a project clean. if the color scheme data isn't in user defaults, you need to make sure to make an API call to get that info